core/worker.cc

```diff
@@ -20,7 +20,9 @@
 int num_workers = 0;
 std::thread worker_threads[MAX_WORKERS];
 Worker *volatile workers[MAX_WORKERS];
-thread_local Worker ctx;
+
+// See worker.h
+__thread Worker ctx;
```

Comment:
worker.h has some brief comments on it. One concern is __thread is a gcc extension, but seems universally supported by other compilers as well.
```
// NOTE: Do not use "thread_local" here. It requires a function call every time
// it is accessed. Use __thread instead, which incurs minimal runtime overhead.
// Worker must be "trivial" type.
```